### DISCUSSION TITLE
Mastodon for Anxious People

### DISCUSSION THREAD

**Discussion by sbeitzel** [chucker/Mastonaut#55] [Ideas]
So, one of the cool things about Mastodon is that it's easy to say, at compose time, that attachments should start out hidden, or even double-hidden if you think about how content warnings work. (Click once to reveal the actual content warning, click again to reveal the attachment.)

I love this.

It's just too bad that, as a *reader* I can't say, "You know what? Don't show me any images by default. Seeing any attachment should require me actively clicking on something, to indicate that yes, I actually want that thing in my eyes."

I once wrote a Twitter client that did just that, and I have to say it made Twitter ever so much more restful. Instead of suffering a zillion messages that consist solely of screenshots of text that would just produce anxiety, I could see what people are actually saying. And if the surrounding text was convincing, I could click through to see what the accompanying image was.

Interestingly, because of the way Mastonaut populates the list of statuses, if you click on the show/hide button to hide an image and then scroll that status up or down so it's no longer in view, then scroll it back in view, the visibility goes back to being whatever the poster set it to be. I think it'd be kind of cool to provide a global preference for, "Hide all attachments by default / Respect poster's visibility setting."

**Comment by fatfingers23** [comment]
Hey @chucker, I know it's not your mess, but could we get these comments removed? Maybe it'll stop. Thanks!

**Reply by chucker** [reply]
Hey, I woke up to this comment and I think GH has already resolved it. :)
